packages/expo-modules-core/ios/Core/Classes/ClassDefinition.swift

```diff
@@ -27,6 +32,7 @@ public final class ClassDefinition: ObjectDefinition {
     self.name = name
     self.constructor = elements.first(where: isConstructor) as? AnySyncFunctionDefinition
     self.associatedType = ~AssociatedObject.self
+    self.isSharedRef = AssociatedObject.self is SharedRerAssociatedObject.Type
```

Comment:
I didn't figure out a better way of checking it. I tried adding another constructor with a different constraint (trying to use `AssociatedObject: SharedRef`), but it didn't help. This is probably because `SharedRef` is a generic type.

Reply:
```suggestion
    self.isSharedRef = AssociatedObject.self is SharedRefAssociatedObject.Type
```